x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/execution/PlanExecutor.java

```diff
@@ -66,6 +68,12 @@ public void searchSource(Configuration cfg, String sql, List<SqlTypedParamValue>
             if (exec instanceof EsQueryExec) {
                 EsQueryExec e = (EsQueryExec) exec;
                 listener.onResponse(SourceGenerator.sourceBuilder(e.queryContainer(), cfg.filter(), cfg.pageSize()));
+            } else if (exec instanceof LocalExec) {
+                listener.onFailure(new PlanningException("Cannot generate a query DSL for an SQL query that either " +
```

Comment:
Shouldn't there be any tests that verify the error messages are being logged when they should?

Reply:
I tried to add unit test for PlanExecutor but it's not easy to create it because of its dependencies. Instead added integ test for `/translate` for those cases.